clib/cPath.mli

```diff
@@ -0,0 +1,18 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(* To be made opaque one day *)
+type t = string
```

Comment:
Can't you make it opaque right now? (you are introducing the type, nobody is relying on its definition)

Reply:
I tried that some time ago, was far from trivial, and I was using dune `Path.t` API as a reference.

So indeed that is a good thing to do, whether to do in this PR or later I don't mind; but would introduce quite a few `Path.from_string`.

Reply:
Made it `private`, so it is a good start IMHO [and Path.t -> string is not so risky IMHO], it is a bit annoying to use, but hopefully when we consolidate the loadpath handling the current api will be much improved.